src/status.c

```diff
@@ -0,0 +1,28 @@
+#include "status.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+static Status global_status = {.code = STATUS_CODE_OK, .caller = NULL, .message = "" };
```

Comment:
According to our style guide, ``static`` variables in the global file context should be prefixed with ``s_``.